### PR TITLE
add more support for vmware

### PR DESCRIPTION
**Pull request by Jason-ZW**
related: #2468 
```
- vmware/
     |- rancheros.iso
     |- rancheros-autoformat.iso
     |- rancheros.vmdk
     |- initrd
     |- rootfs.tar.gz
```
- [x] add vmware autoformat iso
- [x] add vmware noformat iso
- [x] add vmware initrd
- [x] add vmware rootfs
- [x] add vmawre vmdk

### CODE REVIEW

File: Dockerfile.dapper [hunk 1]
```diff
@@ -79,6 +79,7 @@ ARG SYSTEM_DOCKER_VERSION=17.06-ros4
 ARG SYSTEM_DOCKER_URL_amd64=https://github.com/rancher/os-system-docker/releases/download/${SYSTEM_DOCKER_VERSION}/docker-amd64-${SYSTEM_DOCKER_VERSION}.tgz
 ARG SYSTEM_DOCKER_URL_arm64=https://github.com/rancher/os-system-docker/releases/download/${SYSTEM_DOCKER_VERSION}/docker-arm64-${SYSTEM_DOCKER_VERSION}.tgz
 
+ARG VMWARE_APPEND
```
Comment (niusmallnan) [inline Dockerfile.dapper]: There is no need to expose `VMWARE_APPEND`.

File: Dockerfile.dapper [hunk 2]
```diff
@@ -116,6 +117,7 @@ ENV BUILD_DOCKER_URL=BUILD_DOCKER_URL_${ARCH} \
     SYSTEM_DOCKER_URL_amd64=${SYSTEM_DOCKER_URL_amd64} \
     SYSTEM_DOCKER_URL_arm64=${SYSTEM_DOCKER_URL_arm64} \
     VMWARE_AUTOFORMAT=${VMWARE_AUTOFORMAT} \
+    VMWARE_APPEND=${VMWARE_APPEND} \
```
Comment (niusmallnan) [inline Dockerfile.dapper]: Same here.

Add `VMWARE_APPEND` to `ENV DAPPER_ENV ... VMWARE_APPEND`

File: scripts/vmware-release [hunk 3]
```diff
@@ -8,13 +8,81 @@ touch .make-vmware
 source ./scripts/version
 ./scripts/release
 
-ISO_NAME="rancheros-vmware.iso"
+ROOT_FS_NAME="rootfs-vmware.tar.gz"
```
Comment (niusmallnan) [inline scripts/vmware-release]: Do we need the `vmware` suffix here?

File: scripts/vmware-release [hunk 4]
```diff
@@ -8,13 +8,81 @@ touch .make-vmware
...
+else
+    ISO_NAME="rancheros.iso"
+fi
+
+if [ "$VMWARE_APPEND" != "" ]; then
+    echo "--append ${VMWARE_APPEND}"
+    APPEND_PARAM="--append \"${VMWARE_APPEND}\""
+fi
+
+mkdir -p /tmp/dist/openstack/latest && mv ./dist/* /tmp/dist/
```
Comment (niusmallnan) [inline scripts/vmware-release]: openstack?

File: scripts/vmware-release [hunk 5]
```diff
@@ -8,13 +8,81 @@ touch .make-vmware
...
+[ -f ./dist/vmware/success ]
+
+echo Converting ./dist/${VMDK_NAME}
+qemu-img convert -f qcow2 -O vmdk ./dist/vmware/vmdk.img ./dist/artifacts/${VMDK_NAME}
+
+cp ./dist/vmware/artifacts/rancheros.iso ./dist/artifacts/${ISO_NAME} && \
+    cp ./dist/vmware/artifacts/rootfs.tar.gz ./dist/artifacts/${ROOT_FS_NAME} && \
+    cp ./dist/vmware/artifacts/${INITRD} ./dist/artifacts/${INITRD}
+
+cat > ./dist/publish_gss_latest.sh << EOF
```
Comment (niusmallnan) [inline scripts/vmware-release]: Please refer to this: https://github.com/rancher/os/blob/master/scripts/release#L51-L70

File: Dockerfile.dapper (hunk 1 above)
Comment (Jason-ZW) [reply Dockerfile.dapper]: Already modified

File: Dockerfile.dapper (hunk 2 above)
Comment (Jason-ZW) [reply Dockerfile.dapper]: Already modified

File: scripts/vmware-release (hunk 3 above)
Comment (Jason-ZW) [reply scripts/vmware-release]: Already modified

File: scripts/vmware-release (hunk 5 above)
Comment (Jason-ZW) [reply scripts/vmware-release]: Already modified

**Comment by niusmallnan** [comment]
LGTM

**Pull request merged**